utils/nctl/sh/scenarios/upgrade_after_emergency_upgrade_test_pre_1.5.sh

```diff
@@ -0,0 +1,369 @@
+#!/usr/bin/env bash
+
+source "$NCTL"/sh/utils/main.sh
+source "$NCTL"/sh/views/utils.sh
+source "$NCTL"/sh/assets/upgrade.sh
+source "$NCTL"/sh/scenarios/common/itst.sh
+source "$NCTL"/sh/node/svc_"$NCTL_DAEMON_TYPE".sh
+
+# Exit if any of the commands fail.
+set -e
+
+#######################################
+# Runs an integration tests that performs an emergency restart on the network,
+# and then also performs a regular upgrade after.
+# It also simulates social consensus on replacing the original validators (nodes 1-5)
+# with a completely new set (nodes 6-10).
+#
+# Arguments:
+#   `timeout=XXX` timeout (in seconds) when syncing. Default=300 seconds.
+#   `version=X_Y_Z` new protocol version to upgrade to. Default=2_0_0.
+#   `version2=X_Y_Z` new protocol version to upgrade to. Default=3_0_0.
+#######################################
+function main() {
+    log "------------------------------------------------------------"
+    log "Upgrade after emergency upgrade test begins"
+    log "------------------------------------------------------------"
+
+    do_prepare_stage
+
+    # Start the network
+    do_start_network
+    # 0. Await for the completion of the genesis era
+    do_await_genesis_era_to_complete
+    # 1. Send batch of Wasm deploys
+    do_send_wasm_deploys
+    # 2. Send batch of native transfers
+    do_send_transfers
+    # 3. Wait until they're all included in the chain.
+    do_await_deploy_inclusion '1'
+    # 4. Stop the network for the emergency upgrade.
+    do_stop_network
+    # 5. Prepare the nodes for the emergency upgrade.
+    do_prepare_upgrade
+    # 6. Restart the network (start both the old and the new validators).
+    do_restart_network
+    # 7. Wait for the network to upgrade.
+    do_await_network_upgrade
+    # 8. Send batch of Wasm deploys
+    do_send_wasm_deploys
+    # 9. Send batch of native transfers
+    do_send_transfers
+    # 10. Wait until they're all included in the chain.
+    do_await_deploy_inclusion '6'
+    # 11. Prepare the second upgrade.
+    do_upgrade_second_time
+    # 12. Await the second upgrade.
+    do_await_second_upgrade
+    # 13. Reset node 1 to sync from scratch.
+    do_reset_node_1
+    # 14. Wait until node 1 syncs back to genesis.
+    await_node_historical_sync_to_genesis '1' "$SYNC_TIMEOUT_SEC"
+    # 15. Run Health Checks
+    # ... restarts=26: due to nodes being stopped and started; node 1 3 times, nodes 2-5 2 times,
+    # ................ node 6-10 3 times (start at 1.4.8, restart to 1.4.7 for sync, then 2 upgrades)
+    # ... errors=ignore: disabled due to non-deterministic error messages "could not send response
+    # .................. to request down oneshot channel"
+    source "$NCTL"/sh/scenarios/common/health_checks.sh \
+            errors='ignore' \
```

Comment:
This is a bit risky, imho, as other errors may be unnoticed. What do you think about degrading this `error` to `warn`?

@marc-casperlabs can you comment?